handler/src/main/java/io/netty/handler/ssl/OpenSslContext.java

```diff
@@ -509,32 +500,21 @@ static long toBIO(PrivateKey key) throws Exception {
         }
         ByteBuf buffer = Unpooled.directBuffer();
         try {
-            buffer.writeBytes(BEGIN_PRIVATE_KEY);
-            ByteBuf wrappedBuf = Unpooled.wrappedBuffer(key.getEncoded());
-            final ByteBuf encodedBuf;
-            try {
-                encodedBuf = Base64.encode(wrappedBuf, true);
-                try {
-                    buffer.writeBytes(encodedBuf);
-                } finally {
-                    zerooutAndRelease(encodedBuf);
-                }
-            } finally {
-                zerooutAndRelease(wrappedBuf);
+            PemPrivateKey pem = PemPrivateKey.toPEM(key);
+            buffer.writeBytes(pem.content().slice());
```

Comment:
we may also want have the new classes that construct the PEM cert/key allocate direct by default.